python/mcap-protobuf-support/mcap_protobuf/reader.py

```diff
@@ -1,17 +1,27 @@
-"""the :py:mod:`mcap_protobuf.reader` module provides high-level protobuf message reading
-capability. For more low-level control, consider using the underlying
-:py:class:`mcap_protobuf.decoder.Decoder` and :py:class:`mcap.reader.McapReader` objects
-directly.
+"""the :py:mod:`mcap_protobuf.reader` module is deprecated. For similar functionality,
+instantiate the :py:class:`mcap.reader.McapReader` with a
+:py:class:`mcap_protobuf.decoder.DecoderFactory` instance, eg:
+
+>>> from mcap.reader import make_reader
+>>> from mcap_protobuf.decoder import DecoderFactory
+>>> reader = make_reader(open("proto.mcap", "rb"), decoder_factories=[DecoderFactory()])
+>>> for schema_, channel_, message_, proto_msg in reader.iter_decoded_messages():
```

Comment:
What is `proto_msg`? Is that something that would have a meaning to python users?

Why are `schema_, channel_, message_,` variables part of the returned set of variables for each iteration? Could all of these instead by in the "message event" and within that you have the `log_time, publish_time` etc?

Reply:
> What is proto_msg? Is that something that would have a meaning to python users?

It's the protobuf message, or decoded message content in other words.

> Why are schema_, channel_, message_, variables part of the returned set of variables for each iteration? Could all of these instead by in the "message event" and within that you have the log_time, publish_time etc?

There's information in the schema, channel, and message records which can be useful to the caller on every iteration. Hypothetically we could yield a single object here which contains some fields from each, but that requires either just including all fields of all records or selecting which fields might be interesting to all users. This seems like extra effort for little gain, also slightly less performant because we need to construct a new object and copy a bunch of fields instead of just yielding references to the underlying records.